### PR TITLE
Finetune performance

### PR DESCRIPTION
**Pull request by bzz**
This is new, more performant sharding implementation leveraging order by  `repartitionAndSortWithinPartitions`

### CODE REVIEW

**Comment by bzz** [comment]
Ready to be merged, \cc @smola for review

File: src/main/scala/tech/sourced/swivel/SparkPrepDriver.scala [hunk 1]
```diff
@@ -147,21 +150,23 @@ object SparkPrep {
   }
 
   def generateCooccurrence(ids: Array[Int], windowSize: Int) = {
+    //https://github.com/src-d/swivel-spark-prep/issues/5
+    //Re-implement using fastutils.ObjectIntMap
```
Comment (ajnavarro) [inline src/main/scala/tech/sourced/swivel/SparkPrepDriver.scala]: These comments are TODOs? if yes, add the TODO key.

File: src/main/scala/tech/sourced/swivel/SparkPrepDriver.scala [hunk 2]
```diff
@@ -116,9 +119,11 @@ object SparkPrep {
     * @param wordToIdVar
     * @return
     */
-  def wordsToIds(rdd: RDD[Array[String]], wordToIdVar: Broadcast[Map[String, Int]]) = {
-    rdd.map(line => {
-        line.flatMap { // flat is important, skips OOV
+  def wordsToIds(rdd: RDD[Array[String]], wordToIdVar: Broadcast[Map[String, Int]]): RDD[Array[Int]] = {
+    //https://github.com/src-d/swivel-spark-prep/issues/5
+    //Re-implement using RDD[fastutils.IntArrayList]
```
Comment (ajnavarro) [inline src/main/scala/tech/sourced/swivel/SparkPrepDriver.scala]: These comments are TODOs? if yes, add the TODO key.

File: src/test/scala/tech/sourced/swivel/medium/CoocsMatrixMediumTest.scala [hunk 3]
```diff
@@ -85,14 +85,13 @@ class CoocsMatrixMediumTest extends FunSuite with SharedSparkContext {
...
+    val coocs = SparkPrep.mergeCoocsAndShard2(
+      SparkPrep.buildCooccurrenceMatrix(sc.parallelize(tokenLines), wordWindow, wordToIdVar),
       numShards
-    )
-      .collect()
+    ).collect()
 
     // then
-    assert(coocs.length == expectedCoocs.length)
+    //assert(coocs.length == expectedCoocs.length)
```
Comment (ajnavarro) [inline src/test/scala/tech/sourced/swivel/medium/CoocsMatrixMediumTest.scala]: leftover?

**Review by ajnavarro — CHANGES_REQUESTED** [review]

**Comment by bzz** [comment]
@ajnavarro thank for prompt review! Feedback addressed in dc82e9e 
Please let me know if there is anything left before merging.

**Review by ajnavarro — APPROVED** [review]

**Review by smola — APPROVED** [review]

**Comment by bzz** [comment]
@smola @ajnavarro thanks for taking care of it!

**Pull request merged**